src/ErrorBoundary.tsx

```diff
@@ -42,10 +43,26 @@ function emailDevelopers(error: Error, logs: string | undefined) {
 function SessionExpiredDialog() {
   const etebase = useCredentials()!;
   const dispatch = useDispatch();
+  const [fetchFailed, setFetchFailed] = React.useState(false);
   const [password, setPassword] = React.useState("");
   const [errorPassword, setErrorPassword] = React.useState<string>();
 
-  return (
+  React.useEffect(() => {
+    const fetch = async () => {
+      try {
+        await etebase.fetchToken();
+
+        dispatch(login(etebase));
+        dispatch(popError());
+      } catch (e) {
+        setFetchFailed(true);
```

Comment:
Fetch can fail for more than just Unauthorized. If it fails with Unauthorized we should do this, otherwise we should show a temporary error dialog with the error. I guess it's a rare case anyway though. I'll merge it regardless, but just noting it here for the future.

Reply:
Oh yeah I thought about it for a sec and I forgot later. I think it's a one line fix, just have to `addError` so it goes back to ErrorBoundary. I'll make a quick PR

Reply:
Not sure about that, because it could potentially cause infinite loops. I don't know if/what is the easy solution tbh. Maybe better to just leave it.

Reply:
The only case it would cause an infinite loop is if it was an UnauthorizedError, because it's the only error using that dialog, and that's the only case we won't send it back because we prompt for the password.

One question I had, is how does th sync work when you dispatch login. Is it one way server -> client? Because that would make the user lose his local changes.

Reply:
Could you rephrase your question? I don't understand what you mean. Why would it make the user lose his local changes? The login function should only update the stored etebase session.

Reply:
When I tested my commit, I noticed that "Pulling changes" appears in the Note List after the password is entered. My question is just: is this a one-way server->client pull or does it also merge local changes?

Reply:
It should just trigger a normal sync.

Reply:
Oh OK then :+1: